app/src/main/java/com/nut/kiosk/ui/MainActivity.kt

```diff
@@ -68,9 +72,48 @@ class MainActivity : AppCompatActivity(), View.OnTouchListener {
 
     private val compositeDisposable = CompositeDisposable()
 
+    private lateinit var usbPermissionIntent: PendingIntent
+    private lateinit var usbManager: UsbManager
+
+    private val usbReceiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            if (ACTION_USB_PERMISSION == intent.action) {
+                synchronized(this) {
+                    val usbDevice = intent.getParcelableExtra<UsbDevice>(UsbManager.EXTRA_DEVICE)
+                    usbManager.requestPermission(usbDevice, usbPermissionIntent)
```

Comment:
you sending permission request on receiver.
but receiver is not working until you got a permission.

so, you should request permission first.
and then onReceive() method you could do something you want.

Reply:
and there is no permission callback when user allow it or deny it.
refer to the way to implement it below.
https://developer.android.com/training/permissions/requesting.html